Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -198,32 +198,44 @@ namespace AzToolsFramework
 
         AZStd::unique_ptr<AZ::Entity> Instance::DetachEntity(const AZ::EntityId& entityId)
         {
-            EntityAlias entityAliasToRemove;
-            auto instanceToTemplateEntityIdIterator = m_instanceToTemplateEntityIdMap.find(entityId);
-
-            if (instanceToTemplateEntityIdIterator != m_instanceToTemplateEntityIdMap.end())
+            if (m_containerEntity && m_containerEntity->GetId() == entityId)
             {
-                entityAliasToRemove = instanceToTemplateEntityIdIterator->second;
-                if (m_entityIdInstanceRelationship == EntityIdInstanceRelationship::OneToOne)
+                return DetachContainerEntity();
+            }
+            else
+            {
+                EntityAlias entityAliasToRemove;
+                auto instanceToTemplateEntityIdIterator = m_instanceToTemplateEntityIdMap.find(entityId);
+
+                if (instanceToTemplateEntityIdIterator != m_instanceToTemplateEntityIdMap.end())
                 {
-                    if (!UnregisterEntity(entityId))
+                    entityAliasToRemove = instanceToTemplateEntityIdIterator->second;
+                    if (m_entityIdInstanceRelationship == EntityIdInstanceRelationship::OneToOne)
                     {
-                        AZ_Error("Prefab", false, "An owning instance couldn't be found corresponding to the entity requested to be detached");
-                        return nullptr;
+                        if (!UnregisterEntity(entityId))
+                        {
+                            AZ_Error(
+                                "Prefab",
+                                false,
+                                "An owning instance couldn't be found corresponding to the entity requested to be detached");
+                            return nullptr;
+                        }
+                    }
+                    else
+                    {
+                        [[maybe_unused]] bool isEntityRemoved =
+                            m_templateToInstanceEntityIdMap.erase(entityAliasToRemove) && m_instanceToTemplateEntityIdMap.erase(entityId);
```

Comment:
question: Should these `erase` calls be explicitly checking `m_templateToInstanceEntityIdMap.erase(entityAliasToRemove) != m_templateToInstanceEntityIdMap.end()` and `m_instanceToTemplateEntityIdMap.erase(entityId) != m_instanceToTemplateEntityIdMap.end()`? What is it they return if the `erase` doesn't succeed?

Reply:
I believe this was just a formatting change for this PR, but good question. I checked this and the erase returns the number of elements that were actually erased. I'll change the code to check against 0 to make it more clear.

Reply:
Aah I see, thanks for updating, much clearer now 👍